dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImplIntegrationTest.java

```diff
@@ -0,0 +1,130 @@
+    /**
+     * Method to test: {@link FileAssetAPIImpl#findFileAssetsByFolder(Folder, User, boolean)}
+     * When: Create two Folder with two files each one, grant permission over folders and files
+     * and try to get the files from the first folder
+     * Should: return only the two files into the first folder
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void shouldReturnTwoFiles() throws IOException, DotDataException, DotSecurityException {
+        final Role backEndUserRole = APILocator.getRoleAPI().loadBackEndUserRole();
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Folder folder1 = new FolderDataGen().site(host).nextPersisted();
+        final Contentlet fileAsset1 = createFileAsset(folder1, "text1", ".txt");
+        final Contentlet fileAsset2 = createFileAsset(folder1, "text2", ".txt");
+
+        final Folder folder2 = new FolderDataGen().site(host).nextPersisted();
+        final Contentlet fileAsset3 = createFileAsset(folder2, "text1", ".txt");
+        final Contentlet fileAsset4 = createFileAsset(folder2, "text2", ".txt");
+
+        this.addPermission(role, folder1, folder2, fileAsset1, fileAsset2, fileAsset3, fileAsset4);
+        final List<FileAsset> files = fileAssetAPIImpl.findFileAssetsByFolder(folder1, user, false);
+
+        assertEquals(2, files.size());
+
+        final List<String> filesInodes = files.stream().map((file) -> file.getInode()).collect(Collectors.toList());
+
+        assertTrue(filesInodes.contains(fileAsset1.getInode()));
+        assertTrue(filesInodes.contains(fileAsset2.getInode()));
+    }
+
+    /**
+     * Method to test: {@link FileAssetAPIImpl#findFileAssetsByFolder(Folder, User, boolean)}
+     * When: Create one Folder with one files, and try to get the files from the first folder with a user without permission
+     * Should: return a empty List
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void userWithoutPermission() throws IOException, DotDataException, DotSecurityException {
+        final Role backEndUserRole = APILocator.getRoleAPI().loadBackEndUserRole();
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Folder folder1 = new FolderDataGen().site(host).nextPersisted();
+        createFileAsset(folder1, "text1", ".txt");
+
+        final List<FileAsset> fileAssetsByFolder = fileAssetAPIImpl.findFileAssetsByFolder(folder1, user, false);
+
+        assertTrue(fileAssetsByFolder.isEmpty());
+    }
+
+    private Contentlet createFileAsset(
```

Comment:
Cool, I would put this in another more reusable place

Reply:
done https://github.com/dotCMS/core/pull/18225/commits/b3221ca6fdad2fc09e713c11a6b62f4339571aac#diff-a44b1746f41918b71f5f2d06d7cbc482R39